diagnostic_rt_qpcr/libraries/diagnosticrtqpcrcompositions/source.rb

```diff
@@ -15,25 +22,65 @@ module DiagnosticRTqPCRCompositions
   #
   # @param operations [OperationList]
   # @return [void]
-  def build_stripwell_master_mix_compositions(operations:)
+  def build_stripwell_primer_probe_compositions(operations:)
+    all_inputs = []
     operations.each do |op|
-      primer_mixes = []
-      op.input_array(PRIMER_PROBE_MIX).each do |fv|
-        primer_mixes += fv.collection.parts
-      end
-
-      compositions = []
-
-      primer_mixes.each do |primer_mix|
-        composition = build_modified_master_mix_composition(
-          primer_mix: primer_mix,
-          program_name: op.temporary[:options][:program_name]
-        )
-        compositions.append(composition)
+      options = op.temporary[:options]
+      program_name = options[:program_name]
+      
+      stripwells = find_stripwells(sample_names: options[:sample_names],
+                                   stripwell_ot: options[:object_type],
+                                   dimensions: op.temporary[:microtiter_plate].collection.dimensions,
+                                   all_inputs: all_inputs)
+      unless stripwells.present?
+        op.error(ProtocolError, 'No enough parts in inventory')
+        next
       end
-
+      
+      parts_by_row = convert_to_parts_by_row(stripwells: stripwells)
+      
+      compositions = setup_composition_parts(parts_by_row: parts_by_row,
+                                             program_name: program_name)
       op.temporary[:compositions] = compositions
+      all_inputs.push(stripwells.flatten)
+    end
+    all_inputs.flatten
+  end
+  
+  # Converts to parts by row format.  Mostly to allow for easy conversion
+  # if the primer probes ever start coming in as individual parts and
+  # not stripwells
+  def convert_to_parts_by_row(stripwells: stripwells)
+    stripwells.map { |stripwell| stripwell.parts }
+  end
+  
+  def setup_composition_parts(parts_by_row:, program_name:)
+    compositions = []
+    parts_by_row.each do |parts|
+      parts.each do |part|
+        compositions.push(build_primer_probe_compositions(part: part,
+                                        program_name: program_name))
+      end
+    end
+    compositions
+  end
+  
+  def find_stripwells(sample_names:, stripwell_ot:, dimensions:, all_inputs:)
+    ordered_sample_names = Array.new(dimensions[0]/(sample_names.length), sample_names).flatten
+    stripwells = []
+    ordered_sample_names.each do |name|
+      possible_items = Item.where(sample: Sample.find_by_name(name))
+                           .to_ary
+                           .reject{ |part| part unless part.is_part}
```

Comment:
As @bjkeller would say: 🐔 🐔 🐔

Reply:
Pieces parts

Reply:
What on earth is going on here?

Reply:
`reject` block should be something that evaluates to true or false. I would say, rewrite this as 
```
.reject { |item| item.is_part }
```
since you know its an item but not if its a part. If you want the opposite behavior, do 
```
.select { |item| item.is_part }
```
or even better do 
```
.select(&:is_part)
```